File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/CartItem.java
package com.sdl.ecommerce.api.model;

/**
 * Cart Item
 *
 * @author nic
 */
public interface CartItem {

    // TODO: Refactor this into a abstract interface where a number of cart item types can be added, such as ProductItem, PromoItem

    /**
     * Get the accumulated price for the cart item.
     * @return price
     */
    ProductPrice getPrice();

    /**
     * Get product.
     * @return product
     */
    Product getProduct();

    /**
     * Get quantity of the current product
     * @return quantity
     */
    int getQuantity();
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.OData/Model/CartItem.cs
using SDL.ECommerce.Api.Model;

namespace SDL.ECommerce.OData
{
    public partial class CartItem : ICartItem
    {
        IProductPrice ICartItem.Price
        {
            get
            {
                return this.Price;
            }
        }

        IProduct ICartItem.Product
        {
            get
            {
                return this.Product;
            }
        }

        int ICartItem.Quantity
        {
            get
            {
                return (int)this.Quantity;
            }
        }
    }
}
